### ISSUE TITLE
Integrate AmgX with OpenFOAM

### ISSUE THREAD

**Issue by irfanms92** [NVIDIA/AMGX#102]
Hi,
Please let me know if anyone has successfully integrated AmgX library with OpenFOAM.

Regards,
Irfan

**Comment by klausbu** [comment]
Yes, see "Integrating OpenFOAM and GPUs using amgX" by Rathnayake, T 

You could use "PETSc4FOAM" (A Library to plug-in PETSc into the OpenFOAM Framework) and extend it with the "AmgXWrapper"

**Comment by wo315** [comment]
Thank you

**Comment by sbna** [comment]
Hi, we extended the petsc4foam library to support AMGx as well, thanks to @mattmartineau. It is still in dev but if someone is interested...

**Comment by marsaev** [comment]
@mattmartineau do you want make it more visible for users about petsc4foam AMGX integration?

**Comment by marsaev** [comment]
@Dcn303 related paper: http://www.scs.org/wp-content/uploads/2017/06/44_Final_Manuscript.pdf

**Comment by GeBao2644** [comment]
I also encountered the same problem, please do you solve it?

**Comment by Nitish25200** [comment]
> I also encountered the same problem, please do you solve it?

Have you solved your problem

**Comment by Niti2000** [comment]
> Hi, Please let me know if anyone has successfully integrated AmgX library with OpenFOAM.
> 
> Regards, Irfan

Hey, kindly help me , in integration of AMGX with OpenFOAM 
TIA

**Comment by jyluan** [comment]
OpenFOAM-amgx4foam-AmgX-GPU. This packaging seems to be the most concise. If PETSc is used, it will become OpenFOAM-petsc4foam-PETSc-AmgxWrapper-AmgX-GPU (the new version of PETSc supports direct connection with AmgX, so AmgxWrapper can be omitted). In this case, will more packaging hinder the acceleration?
